wagtail/admin/utils.py

```diff
@@ -0,0 +1,26 @@
+from django.conf import settings
+
+
+def get_admin_base_url(context=None):
+    """
+    Gets the base URL for the wagtail admin site. This is set in `settings.WAGTAILADMIN_BASE_URL`,
+    which was previously `settings.BASE_URL`.
+    If setting is omitted and this is called in a request context, falls back to
+    `request.site.root_url` or next the host_name of the request.
+    """
+
+    admin_base_url = getattr(settings, "WAGTAILADMIN_BASE_URL", None)
+    if admin_base_url is None:
+        admin_base_url = getattr(settings, "BASE_URL", None)
```

Comment:
We may want to add a deprecated warning here, but only if `WAGTAILADMIN_BASE_URL` is `None` and `BASE_URL` is not `None`.

See `wagtail/utils/deprecation.py` for the utils we use for this.

This would probably be a two release warning, so I recommend you use `RemovedInWagtail50Warning`, you can search through the code for other examples.